posydon/binary_evol/CE/step_CEE.py

```diff
@@ -814,4 +826,92 @@ def CEE_simple_alpha_prescription(
+
+                    Mejected_donor = Mejected_donor / const.Msun # to go from cgs to solar masses again
+                    Mejected_comp = 0.0
+
+                    if Mejected_donor > m1_i  - mc1_i:
+                        Mejected_donor = (m1_i  - mc1_i) -0.01 # at least this value of envelope is left.
+                        warnings.warn("Mejected_donor in normal CEE is found to be more that the initial envelope. Reduced to initial_envelope - 0.01 Msun")
+
+                else: # in double_CE
+
+                    # Assuming that the ratio of ejected mass of each (common) envelope
+                    # is the same as the ratio of the initial envelope masses:
+                    # the root of the mass lost for each is the solution of a quadratic algebraic equation
+
+                    WF = (m1_i  - mc1_i)/ (m2_i - mc2_i) # weight factor:
+                                                        # = Mdonor,envelope / Mcomp,envelope == Mejected_donor / Mejected_comp
+
+                    # Mejected_comp = Mejected_donor / WF
+                    # A*Mejected_donor^2 + B*Mejected_donor + C = 0
+
+                    A = alpha_CE / (WF*2.*separation_postCEE)
+                    B = ( ( - alpha_CE*m1_i * const.Msun/ (WF*2.*separation_postCEE)) - (alpha_CE*m2_i * const.Msun/ (2.*separation_postCEE)) \
+                        - (m1_i * const.Msun/ (lambda1_CE*radius1 * const.Rsun)) - (m2_i * const.Msun/ (WF*lambda2_CE*radius2 * const.Rsun)))
+                    C = alpha_CE * (m1_i * const.Msun * m2_i * const.Msun)/2. * (1./separation_postCEE - 1./separation_i)
+                    roots = np.roots([A,B,C])
+                    if min( np.real(roots[0]), np.real(roots[1]) ) < 0.:
+                        root = max( np.real(roots[0]), np.real(roots[1]) ) # if one root is negative, the other is considered the physical solution
+                    elif max( np.real(roots[0]), np.real(roots[1]) ) > m1_i  - mc1_i: # if one root is more positive than the Menvelope that we allow to ejected
+                                                                                     # then the other is considered the physical solution
+                        root = min( np.real(roots[0]), np.real(roots[1]) )
+
+                    Mejected_donor = root  / const.Msun # to go from cgs to solar masses again
+                    Mejected_comp = Mejected_donor / WF
+
+                    if verbose:
+                        print("The roots are: ", roots / const.Msun, "in Msun")
+                        print("so Mejecta_donor = ", Mejected_donor, "in Msun compared to its initial envelope =",  m1_i  - mc1_i)
+                        print("so Mejecta_comp = ", Mejected_comp, "in Msun compared to its initial envelope =",  m2_i  - mc2_i)
+
+                    if not ( (Mejected_donor <= m1_i  - mc1_i) and (Mejected_comp <= m2_i  - mc2_i) ):
```

Comment:
Again, the units on both sides of the comparison look to be different.
As you use the original envelope a lot, it might be better to even store this in a variable.